test/parallel/test-fs-utimes.js

```diff
@@ -164,7 +164,26 @@ runTest(new Date('1982-09-10 13:37'), new Date('1982-09-10 13:37'), function() {
 });
 
 
+// Ref: https://github.com/nodejs/node/issues/13255
+if (common.isWindows) {
+  common.refreshTempDir();
+  const path = `${common.tmpDir}/test-utimes-precision`;
+  fs.writeFileSync(path, '');
+
+  // this value would get converted to (double)1713037251359.9998
+  const truncate_mtime = 1713037251360;
+  fs.utimesSync(path, truncate_mtime / 1000, truncate_mtime / 1000);
+  const truncate_stats = fs.statSync(path);
+  assert.strictEqual(truncate_mtime, truncate_stats.mtime.getTime());
+
+  // if this value is cast to `signed long` it gets converted to -2135622133469
```

Comment:
That’s not really a comment explaining what’s happening…

It’s not like `(signed long) 2159345162531 == -2135622133469` – it’s more complicated than that, it definitely relates to how the Windows API deals with timestamps.